src/executor/evaluate/evaluated.rs

```diff
@@ -138,6 +138,13 @@ impl<'a> Evaluated<'a> {
         }
     }
 
+    pub fn unary_factorial(&self) -> Result<Evaluated<'a>> {
+        match self {
+            Evaluated::Literal(v) => v.unary_factorial().map(Evaluated::Literal),
+            Evaluated::Value(v) => v.unary_factorial().map(Evaluated::from),
+        }
```

Comment:
In the `unary_factorial` impl in `data/literal.rs`, it converts `Literal::Number` only into `i64` which can be directly matched to `Value::I64`.
Value or literal flows from less informative type to more informative type, and `Value` has more info than `Literal`.
Current conversion from `BigDecimal` to `i64` really makes sense. However, we don't need to convert back to `BigDecimal` after the calculation has done.
Then, we might not need `Literal::unary_factorial`.
Rather than `Literal::unary_factorial`, we can convert `Literal` into `Value` here.

```rust
match self {                                                                      
    Evaluated::Literal(v) => Value::try_from(v).and_then(|v| v.unary_factorial()),
    Evaluated::Value(v) => v.unary_factorial(),                                   
}.map(Evaluated::from)
```

Reply:
@panarch 

thanks..
it is very beautiful..

modified code.
and, delete literal factorial code